### DISCUSSION TITLE
Using existing schema for requests

### DISCUSSION THREAD

**Discussion by wiedehoeft** [graphql-java-kickstart/graphql-spring-boot#654] [Q&A]
The Graphql-java tools provide the possibility to execute queries on server-side: https://www.graphql-java.com/documentation/v16/execution/

But it is unclear how to use this with graphql-java-kickstart. The existing schema file is parsed and the only need for implementations are the resolver with a `DataFetchingEnvironment`.

Exists part that can be reused for the upper example to use the existing, parsed schema definition for new custom queries? 

We have tried it this way
```
    File schemaFile1 = new File(getClass().getClassLoader().getResource("mainAccount-schema.graphqls").getFile());
    final TypeDefinitionRegistry typeDefinitionRegistry = new SchemaParser().parse(schemaFile1);

    final GraphQLSchema schema = new SchemaGenerator().makeExecutableSchema(typeDefinitionRegistry, RuntimeWiring.newRuntimeWiring().build());
    GraphQL graphQL = GraphQL.newGraphQL(schema)
      .build();

    ExecutionInput executionInput = ExecutionInput.newExecutionInput().query("[...]")
      .build();

    ExecutionResult executionResult = graphQL.execute(executionInput);
    Object data = executionResult.getData();
    List<GraphQLError> errors = executionResult.getErrors();
```
But the result is always empty `{data={getMainAccount=null}}`, after execution with a SpringBootIntegrationTest. This does not surprise, because the classes which implement methods with `dataFetchingEnvironment` are not called.

**Comment by ermadmi78** [comment]
See example of Kotlin GraphQL client for graphql-java-kickstart server [here](https://github.com/ermadmi78/kobby-gradle-example)

**Reply by wiedehoeft** [reply]
Thank you,
the missing link was the injection :) 
Maybe it would be helpful to add this hint in the documentation, that the configured objects are injectable.
